GoogleSignIn/Sources/GIDToken.m

```diff
@@ -55,4 +57,39 @@ - (void)encodeWithCoder:(NSCoder *)encoder {
   [encoder encodeObject:_expirationDate forKey:kExpirationDateKey];
 }
 
+#pragma mark - isEqual
+
+- (BOOL)isEqual:(id)object {
+  if (self == object) {
+    return YES;
+  }
+  if (![object isKindOfClass:[GIDToken class]]) {
+    return NO;
+  }
+  return [self isEqualToToken:(GIDToken *)object];
+}
+
+- (BOOL)isEqualToToken:(GIDToken *)other {
+  return [_tokenString isEqual:other.tokenString] &&
+      [self isTheSameDate:_expirationDate with:other.expirationDate];
+}
+
+- (BOOL)isTheSameDate:(NSDate *)date1
+                 with:(NSDate *)date2 {
+  
+  if (date1 || date2) {
```

Comment:
Might be more intuitive to do something like:

```
if (date1 and date2 are nil) {
  return YES;
}
return [date1 isEqualToDate:date2];
```

Reply:
if (!date1 && !date2) is not working because of the short circuit. Checking at least one is not nil is easier.